test/EspressoSGXTEEVerifier.t.sol

```diff
@@ -0,0 +1,220 @@
+
+        vm.expectRevert(IEspressoSGXTEEVerifier.InvalidQuote.selector);
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+    }
+
+    function testRegisterSignerInvalidAddress() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        address batchPosterAddress = address(0x4C91660a37d613E1Bd278F9Db882Cc5ED2549072);
+
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+
+        vm.expectRevert(IEspressoSGXTEEVerifier.InvalidReportDataHash.selector);
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+    }
+
+    function testRegisterSignerInvalidDataLength() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+
+        // encode adds padding and the length should become incorrect
+        bytes memory data = abi.encode(batchPosterAddress);
+
+        vm.expectRevert(IEspressoSGXTEEVerifier.InvalidDataLength.selector);
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+    }
+
+    function testDeleteRegisteredSigner() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+
+        address batchPosterAddress = address(0xe2148eE53c0755215Df69b2616E552154EdC584f);
+
+        // Convert to bytes (dynamically sized)
+        bytes memory data = abi.encodePacked(batchPosterAddress);
+
+        espressoSGXTEEVerifier.registerSigner(sampleQuote, data);
+        vm.stopPrank();
+
+        vm.startPrank(adminTEE);
+        espressoSGXTEEVerifier.deleteRegisteredSigner(batchPosterAddress);
+        assertEq(espressoSGXTEEVerifier.registeredSigners(batchPosterAddress), false);
+        vm.stopPrank();
+    }
+
+    /**
+     * Test verify quote verifies that if correct quote and report data hash is passed
+     *     then the function does not revert
+     */
+    function testVerifyQuoteValid() public {
+        vm.startPrank(adminTEE);
+        string memory quotePath = "/test/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        espressoSGXTEEVerifier.verify(sampleQuote, reportDataHash);
+        vm.stopPrank();
+    }
+
+    /**
+     * Test verify quote reverts if incorrect header is passed
+     */
+    function testVerifyInvalidHeaderInQuote() public {
```

Comment:
would it be worthwhile to add a test for verification of zero values?

Reply:
What would zero values mean here?

Reply:
bytes calldata rawQuote, bytes32 reportDataHash - either of these are zero? and the verify function should revert then too